datatransfer/impl/graphsync/graphsync_impl.go

```diff
@@ -81,57 +85,40 @@ func (impl *graphsyncImpl) gsReqRecdHook(p peer.ID, request graphsync.RequestDat
 	var resp []graphsync.ExtensionData
 
 	// if this is a push request the sender is us.
-	tid, err := impl.transferIDFromExtension(request)
+	transferData, err := impl.getExtensionData(request)
 	if err != nil {
 		return resp, err
 	}
-
-	extData := graphsync.ExtensionData{
-		Name: ExtensionDataTransfer,
-		Data: nil,
+	raw, _ := request.Extension(ExtensionDataTransfer)
+	respData := graphsync.ExtensionData{Name: ExtensionDataTransfer, Data: raw}
+	resp = append(resp, respData)
+
+	sender := impl.peerID
+	initiator := impl.peerID
+	if transferData.IsPull {
+		// if it's a pull request: the initiator is them
+		initiator = p
 	}
-	
-	chid := datatransfer.ChannelID{ Initiator: impl.peerID, ID: tid}
-	// if a push request, initiator & sender is us, ask for the channel with us as sender
-	if impl.getChannelByIdAndSender(chid, impl.peerID) == datatransfer.EmptyChannelState {
-
-		// otherwise check if it's a pull request: the initiator is them
-		tid, err = impl.transferIDFromExtension(request)
-
-		chid = datatransfer.ChannelID{Initiator: p, ID: tid}
-		// sender is still us
-		if impl.getChannelByIdAndSender(chid, impl.peerID) == datatransfer.EmptyChannelState {
-			return resp, errors.New("could not find push or pull channel")
-		}
+	chid := datatransfer.ChannelID{Initiator: initiator, ID: datatransfer.TransferID(transferData.TransferID)}
+
+	if impl.getChannelByIDAndSender(chid, sender) == datatransfer.EmptyChannelState {
+		return resp, errors.New("could not find push or pull channel")
 	}
-	resp = append(resp, extData)
+
 	return resp, nil
 }
 
-// gsExtended is a small interface used by transferIDFromExtension
+// gsExtended is a small interface used by getExtensionData
```

Comment:
This was here in  order to be usable by both request and response hooks, but now we aren't going to use a response hook. I can get rid of it if desired.